py_experimenter/result_processor.py

```diff
@@ -31,21 +30,26 @@ def __init__(self, config: ConfigParser, use_codecarbon: bool, codecarbon_config
         self._codecarbon_config = codecarbon_config
 
         if config['PY_EXPERIMENTER']['provider'] == 'sqlite':
-            self._dbconnector: DatabaseConnector = DatabaseConnectorLITE(config, self.use_codecarbon, self._codecarbon_config, logger_name)
+            self._dbconnector: DatabaseConnector = DatabaseConnectorLITE(config, self.use_codecarbon, self._codecarbon_config, logger)
         elif config['PY_EXPERIMENTER']['provider'] == 'mysql':
-            self._dbconnector: DatabaseConnector = DatabaseConnectorMYSQL(config, self.use_codecarbon, self._codecarbon_config, credential_path, logger_name)
+            self._dbconnector: DatabaseConnector = DatabaseConnectorMYSQL(
+                config, self.use_codecarbon, self._codecarbon_config, credential_path, logger)
         else:
             raise InvalidConfigError("Invalid database provider!")
 
-    def process_results(self, results: dict) -> None:
+        self._result_fields = utils.get_result_field_names(self._config)
+        self._logtable_fields = utils.extract_logtables(config, table_name)
+
+    def process_results(self, results: Dict) -> None:
         """
         Process results from the experiment and write them to the database. You can call this method, whenever you
         want to write results to the database.
         :param results: Dictionary with result field name and result value pairs.
         """
         if not self._valid_result_fields(list(results.keys())):
             invalid_result_keys = set(list(results.keys())) - set(self._result_fields)
-            raise InvalidResultFieldError(f"Invalid result keys: {invalid_result_keys}")
+            logging.error(f"The resultsfileds `{','.join(invalid_result_keys)}` are invalid.")
```

Comment:
Invalid wrt what? They probably don't exists in the DB and therefore do not match with the experiment configuration at the time the DB was set up. Would be good to have this information given here.

Reply:
I changed this